Tests/NIOTests/ChannelPipelineTest.swift

```diff
@@ -69,6 +69,38 @@ private extension EmbeddedChannel {
 
 class ChannelPipelineTest: XCTestCase {
 
+    class SimpleTypedHandler1: ChannelInboundHandler {
+        typealias InboundIn = NIOAny
+    }
+    class SimpleTypedHandler2: ChannelInboundHandler {
+        typealias InboundIn = NIOAny
+    }
+    class SimpleTypedHandler3: ChannelInboundHandler {
+        typealias InboundIn = NIOAny
+    }
+
+    func testGetHandler() throws {
+        let handler1 = SimpleTypedHandler1()
+        let handler2 = SimpleTypedHandler2()
+        let handler3 = SimpleTypedHandler3()
+        
+        let channel = EmbeddedChannel()
+        try channel.pipeline.addHandlers([
+            handler1,
+            handler2,
+            handler3
+        ]).wait()
+        
+        let result1 = try channel.pipeline.handler(type: SimpleTypedHandler1.self).wait()
+        XCTAssertTrue(result1 === handler1)
+        
+        let result2 = try channel.pipeline.handler(type: SimpleTypedHandler2.self).wait()
+        XCTAssertTrue(result2 === handler2)
+
+        let result3 = try channel.pipeline.handler(type: SimpleTypedHandler3.self).wait()
+        XCTAssertTrue(result3 === handler3)
```

Comment:
nice! We could do some more tests here, like

- two handlers of the same type in the pipeline, verify that we get the first of those
- no handler of the given type there, verify that we get `nil`